### PR TITLE
Feature: start delay

### PR DESCRIPTION
**Pull request by jamesjwarren**
Adds the option to start animation after a specified time delay.

And can also be initialised via the velocityGroup directive:

``` html
<sn-velocity-group data-keyframes="{ '#elem1': { 'keyframes': [{ 'properties': { 'opacity': '1' }, 'options': { 'duration': '1000' } }], 'startDelay': '200' } }"></div>
```

This requires a new format for the keyframes object to allow each animation element in the group to have it's own start delay. This does not break the old format, this directive now accepts either format.

### CODE REVIEW

File: app/js/directives/velocity.js [hunk 1]
```diff
@@ -4,24 +4,64 @@
...
+
+                /**
+                 * Trigger animation, with loop or start delay on initialisation
+                 * @method init
+                 */
+                $scope.init = function init() {
+
+                    // start delay - run animation on init or with start delay
+                    if ($scope.startDelay) {
+                        timer = $timeout($scope.animate, $scope.startDelay);
```
Comment (edoparearyee) [inline app/js/directives/velocity.js:58]: Here you should cancel the current instance of `$timeout` before creating a new instance. `if (timer) { $timeout.cancel(timer) }`

File: app/js/directives/velocity.js [hunk 2]
```diff
@@ -4,24 +4,64 @@
...
+                    }
+                };
+
+                /**
+                 * Clear all listeners during angularjs garbage collection
+                 * @method onDestroy
+                 */
+                $scope.onDestroy = function onDestroy() {
+                    timer.cancel();
+                };
```
Comment (edoparearyee) [inline app/js/directives/velocity.js:72]: Can you call `cancel` on an instance of `$timeout`? 

If you can you also want to check that `timer` is not `undefined` before calling `cancel()` else you will get a error e.g. `if (timer && timer.cancel) { timer.cancel() }`

**Comment by jamesjwarren** [comment]
Should be ready now.

**Comment by edoparearyee** [comment]
:+1:

**Pull request merged**